### DISCUSSION TITLE
How to obtain/output the profile of a variable along a specific direction

### DISCUSSION THREAD

**Discussion by Bigdogcat** [idaholab/moose#25020] [Q&A Modules: General] [answered]
Hi,

Assuming that I have a 2D simulation, how to obtain the value profile of a variable say D along the Y-axis direction (i.e. X=0) via Postprocessor? In other words, how to obtain the value of variable D along the red line as shown below. 

<img width="208" alt="截屏2023-07-22 下午11 17 38" src="https://github.com/idaholab/moose/assets/73391904/c39e6170-65d3-4dbd-ab17-3afcffa66a31">

Many thanks!

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can use the LineValueSampler vectorpostprocessor for this
https://mooseframework.inl.gov/source/vectorpostprocessors/LineValueSampler.html

Guillaume

**Reply by Bigdogcat** [reply]
Thanks Guillaume!!

**Reply by Bigdogcat** [reply]
Hi Guillaume,

May I also ask is it possible to output the sampler results every n time steps (e.g. n = 20) rather than just the default n=1? Thanks very much

Qingpei

**Reply by GiudGiud** [reply]
Hello

Right now it can be done with the controls system. 

It s a little difficult to set up. 
we are working on a simpler system. For now I would just let it execute too often.

Guillaume

**Reply by Bigdogcat** [reply]
Thanks Guillaume. Got it!
